### ISSUE TITLE
Postback not parsing... I think

### ISSUE THREAD

**Issue by rrottier** [moleike/line-bot-sdk#1]
I am trying to use the quickreply postback action.  The quickreply works but I never receive the EventPostback.  Looking at the logging it seems like something is triggering a 400 reply but I can't trace where it is coming from.

 - - [29/Dec/2020:11:19:56 +0000] "POST /webhook HTTP/1.1" 400 - "" "LineBotWebhook/1.0"

EventMessage works fine so it makes me think that maybe something is wrong with the parsing of EventPostback?

**Comment by moleike** [comment]
Hi @rrottier,

**Comment by rrottier** [comment]
Yes, my code for the quick reply is:
```
askChapter :: (MonadIO m, MonadChannel m) => ReplyToken -> m ()
askChapter rt = do
  token <- getToken
  _ <- liftIO $ runLine comp token
  return ()
    where
      welcome = "Roughly what MinnaNoNihongo chapter are you up to? (Don't worry you can always change it later)"
      qr      = QuickReply [QuickReplyButton Nothing (ActionPostback "Chapter5" "5" "Ch 5"),
                            QuickReplyButton Nothing (ActionPostback "Chapter10" "10" "Ch 10"),
                            QuickReplyButton Nothing (ActionPostback "Chapter15" "15" "Ch 15"),
                            QuickReplyButton Nothing (ActionPostback "Chapter20" "20" "Ch 20"),
                            QuickReplyButton Nothing (ActionPostback "Chapter25" "25" "Ch 25"),
                            QuickReplyButton Nothing (ActionPostback "Chapter30" "30" "Ch 30"),
                            QuickReplyButton Nothing (ActionPostback "Chapter35" "35" "Ch 35"),
                            QuickReplyButton Nothing (ActionPostback "Chapter40" "40" "Ch 40"),
                            QuickReplyButton Nothing (ActionPostback "Chapter45" "45" "Ch 45"),
                            QuickReplyButton Nothing (ActionPostback "Chapter50" "50" "Ch 50")
                            ]
      comp    =  replyMessage rt [B.MessageText welcome (Just qr)]
```

**Comment by rrottier** [comment]
Let me know if there is anything else that would help.  I am completely stuck.

**Comment by rrottier** [comment]
Ah, I found the issue.  

```
data Postback = Postback Text PostbackDateTime
  deriving (Eq, Show)

instance FromJSON Postback where
  parseJSON = withObject "Postback" $ \o -> do
    postbackData <- o .: "data"
    params       <- o .: "params"
    return $ Postback postbackData params
```
Should be:
```
data Postback = Postback Text (Maybe PostbackDateTime)
  deriving (Eq, Show)

instance FromJSON Postback where
  parseJSON = withObject "Postback" $ \o -> do
    postbackData <- o .: "data"
    params       <- o .:? "params"
    return $ Postback postbackData params
```
As 'params' is only returned if you use the date picker in the postback action.

**Comment by moleike** [comment]
@rrottier thanks for looking into this! Would you mind submitting a PR?

**Comment by rrottier** [comment]
Doen #2

**Comment by moleike** [comment]
I just published version 0.6.0 in Hackage with the postback fix (also added is the user-agent request header, which should not affect anything). If all it's ok, please close this issue.
